### PR TITLE
[Improvement-3369][api] Introduce monitor, processinstance and queue service interface for clear code

### LINKED ISSUE #3369: [Improvement][api] Introduce interface about api module service
**Describe the question**
Introduce interface about api module service for clearly service module.

**What are the current deficiencies and the benefits of improvement**
- A clear and concise service will help contributor or user to ready the code and locate the function.

**Which version of DolphinScheduler:**
 -[dev]

**Describe alternatives you've considered**
Introduce service interface

### PR DESCRIPTION
**Pull request by chengshiwen**
## What is the purpose of the pull request

*Introduce monitor, processinstance and queue interface for clear code and close pr #3443 which has conflicts and not updated for a long time*

Compared to pr #3443 with **2 vulnerabilities, 15 code smells and 1.4% duplication** half year ago, there are only **17 code smells** in this pr.

This closes #3369

## Brief change log

  - *Add MonitorServiceImpl and change MonitorService to interface*
  - *Add ProcessInstanceServiceImpl and change ProcessInstanceService to interface*
  - *Add QueueServiceImpl and change QueueService to interface*

## Verify this pull request

*(Please pick either of the following options)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

### CODE REVIEW

**Comment by chengshiwen** [comment]
@CalvinKirs Now only 9 code smells
image

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessInstanceServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,741 @@
...
+            putMsg(result, Status.DATA_IS_NULL, Constants.END_TIME);
+            return result;
+        }
+        Date end = DateUtils.stringToDate(endTime);
+        if (start == null || end == null) {
+            putMsg(result, Status.REQUEST_PARAMS_NOT_VALID_ERROR, "startDate,endDate");
+            return result;
+        }
+        if (start.getTime() > end.getTime()) {
+            putMsg(result, Status.START_TIME_BIGGER_THAN_END_TIME_ERROR, startTime, endTime);
```
Comment (CalvinKirs) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessInstanceServiceImpl.java]: It's best not to show magic numbers

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessInstanceServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,741 @@
...
+    }
+
+    /**
+     * add dependent result for dependent task
+     */
+    private void addDependResultForTaskList(List<TaskInstance> taskInstanceList) throws IOException {
+        for (TaskInstance taskInstance : taskInstanceList) {
+            if (taskInstance.getTaskType().equalsIgnoreCase(TaskType.DEPENDENT.toString())) {
+                Result<String> logResult = loggerService.queryLog(
+                        taskInstance.getId(), 0, 4098);
```
Comment (CalvinKirs) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessInstanceServiceImpl.java]: It's best not to show magic numbers

**Review by CalvinKirs — CHANGES_REQUESTED** [review]

**Comment by CalvinKirs** [comment]
There are too many places with magic numbers. This is a problem left over from history. Please correct it

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessInstanceServiceImpl.java (hunk 2 above)
Comment (chengshiwen) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessInstanceServiceImpl.java]: Done

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessInstanceServiceImpl.java (hunk 1 above)
Comment (chengshiwen) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessInstanceServiceImpl.java]: Done

**Review by CalvinKirs — APPROVED** [review]
LGTM

**Pull request merged**